File: SwiftAlgorithm/BOJ/그리디,정렬/11047.swift
//
//  main.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2022/03/14.
//

//11047 동전 0
import Foundation

let input1 = readLine()!.split(separator: " ").map { Int(String($0))! }
let count = input1[0]
var sum = input1[1]

var coins: [Int] = []
var coinCount = 0

for _ in 0..<count {
    let input = Int(readLine()!)!
    coins.append(input)
}

coins.sort(by: >)

for coin in coins {
    if sum >= coin {
        coinCount += (sum / coin)
        sum %= coin
    }
}

print("\(coinCount)")

File: Python/BOJ/그리디/11047.py
//
//  11047.py
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/05.
//

import sys; readLine = sys.stdin.readline

n, k = map(int, readLine().split())
coins = []
for _ in range(n):
  coins.append(int(readLine().rstrip()))

coins.reverse()

result = 0
for coin in coins:
  if coin <= k:
    count = k // coin
    k -= coin * count
    result += count

print(result)
